_config/config.yml

```diff
@@ -27,55 +27,58 @@ Name: snapshot-listeners
 ---
 SilverStripe\CMS\Controllers\CMSMain:
   extensions:
-    - SilverStripe\Snapshots\Listener\Page\CMSMainAction
+    - SilverStripe\Snapshots\Listener\CMSMain\CMSMainActionListener
```

Comment:
Generally agree. The names of these classes went through several changes. :)

A couple points:
* We don't have custom `Context` objects anymore. I thought that was too over-engineered, and made it less extensible. Now it's just one `EventContext` object with arbitrary meta, aside from the required action. Without those context objects, the directories are less populated. `CMSMain`, for instance, has just the handler class.

If we just name it `Handler`, then that presumes there will never be any more CMSMain events. I suppose that's likely to hold true, but with GridField/GraphQL, it does feel nice to group them in the same namespace rather than having `GridFieldAlteration` as a namespace. 

I think more on this one.

Reply:
Have revised to `Handler` and `Listener` classes as suggested. For  GridField and GraphQL, I just created a subdivision of the namespace. `GridField\Action\Handler`.